tests/jetstream.rs

```diff
@@ -247,162 +247,255 @@ fn jetstream_publish() {
+        subjects: Some(vec![
+            "foo".to_string(),
+            "bar".to_string(),
+            "baz".to_string(),
+            "foo.*".to_string(),
+        ]),
+        ..Default::default()
+    })
+    .unwrap();
 
-    js.existing("test2", "consumer3")?;
+    js.stream_info("TEST").unwrap();
 
-    // cleanup
-    let streams: io::Result<Vec<StreamInfo>> = js.list_streams().collect();
+    for _ in 0..10 {
+        js.publish("bar", b"hello js").unwrap();
+    }
 
-    for stream in streams? {
-        let consumers: io::Result<Vec<ConsumerInfo>> =
-            js.list_consumers(&stream.config.name)?.collect();
+    // Create a queue group on "bar" with no explicit durable name, which
+    // means that the queue name will be used as the durable name.
+    let sub1 = js.queue_subscribe("bar", "v0").unwrap();
 
-        for consumer in consumers? {
-            js.delete_consumer(&stream.config.name, &consumer.name)?;
-        }
+    // Since the above JS consumer is created on subject "bar", trying to
+    // add a member to the same group but on subject "baz" should fail.
+    js.queue_subscribe("baz", "v0").unwrap_err();
 
-        js.purge_stream(&stream.config.name)?;
+    // If the queue group is different, but we try to attach to the existing
+    // JS consumer that is created for group "v0", then this should fail.
+    js.queue_subscribe_with_options(
+        "bar",
+        "v1",
+        &SubscribeOptions::new().durable_name("v0".to_string()),
+    )
+    .unwrap_err();
+
+    // However, if a durable name is specified, creating a queue sub with
+    // the same queue name is ok, but will feed from a different JS consumer.
+    let sub2 = js
+        .queue_subscribe_with_options(
+            "bar",
+            "v0",
+            &SubscribeOptions::new().durable_name("other_queue_durable".to_string()),
+        )
+        .unwrap();
 
-        assert_eq!(js.stream_info(&stream.config.name)?.state.messages, 0);
+    let msg = sub1.next().unwrap();
+    msg.ack().unwrap();
+    assert_eq!(msg.data, b"hello js");
 
-        js.delete_stream(&stream.config.name)?;
-    }
+    let msg = sub2.next_timeout(Duration::from_secs(1)).unwrap();
+    msg.ack().unwrap();
+    assert_eq!(msg.data, b"hello js");
 
-    Ok(())
+    sub1.unsubscribe().unwrap();
+    sub2.unsubscribe().unwrap();
 }
 
 #[test]
-fn jetstream_libdoc_test() {
-    let (_s, nc, js) = run_basic_jetstream();
+fn jetstream_flow_control() {
+    let s = util::run_server("tests/configs/jetstream.conf");
+    let nc = nats::connect(&s.client_url()).unwrap();
+    let js = nats::jetstream::new(nc);
 
-    js.add_stream("my_stream").unwrap();
-    nc.publish("my_stream", "1").unwrap();
-    nc.publish("my_stream", "2").unwrap();
-    nc.publish("my_stream", "3").unwrap();
-    nc.publish("my_stream", "4").unwrap();
+    js.add_stream(&StreamConfig {
+        name: "TEST".to_string(),
+        subjects: Some(vec![
+            "foo".to_string(),
```

Comment:
Anyway to allow just vec!["foo", "bar"], etc?

Reply:
Breaking change but yes, empty vector is as meaningful as an optional one imo.